### DISCUSSION TITLE
-Dargs.outext=htm doesn't seem to work?

### DISCUSSION THREAD

**Discussion by JackBaltzer** [dita-ot/dita-ot#4494] [Q&A] [answered]
Greetings, 

I am trying to run a dita command with the parameter ~~`-Dargs.output=htm`~~  `-Dargs.outext=htm` but it seems like something goes wrong with setting the value.
![image](https://github.com/user-attachments/assets/39b9a2ac-3539-4bf8-86de-f676ad3fdd97)
As you can see the value is set to `args=.outext` instead of `args.outext=htm`

I am using this command (with some additional settings afterwards) and I have tried moving the command order around to see if anything new happens, but same result.
![image](https://github.com/user-attachments/assets/7685bd1d-ef22-4efd-8d53-2a12af0af772)

If I don't include the command ~~`-Dargs.output=htm`~~  `-Dargs.outext=htm` then no outext is configured, even thou there are a .opt file with the setting:
![image](https://github.com/user-attachments/assets/b0386869-bc3f-4123-a096-a19df90c2061)

this is the log when I don't use the command line argument.
![image](https://github.com/user-attachments/assets/4dff1d7c-6602-48a7-a9ac-82bbef901a29)

So... what to do? I need the output extension to be .htm but I can only get it to output .html

**Comment by raducoravu** [comment]
@JackBaltzer  that "opt" file is a configuration file specific for the Oxygen XML WebHelp responsive output. Do I understand correctly? 
Ideally this DITA OT discussions list should be for publishing using base plugins which come bundled with the DITA OT like HTML5, XHTML, classic XSL-FO based PDF.

**Reply by JackBaltzer** [reply]
You are probably right, I am not the one working on the actual website, I just have the task of building and pushing it to the server. I included the .opt file because it contains a parameter targeting the same outext, and figured they were releated.

**Comment by jlacour31** [comment]
@JackBaltzer I added an issue to analize the "opt" file problem in Oxygen. As @raducoravu mentioned this isn't a DITA-OT related issue.
Regarding the command line, from what I tested, everything works correctly, please note the correct parameter is `-Dargs.outext=htm` (not `args.output`)
```
log-arg:
     [echo] *****************************************************************
     [echo] * basedir = XXX
     [echo] * dita.dir = XXX
     [echo] * transtype = webhelp-responsive
     [echo] * tempdir = XXX
     [echo] * outputdir = XXX
     [echo] * clean.temp = yes
     [echo] * DITA-OT version = 4.1.2
     [echo] * XML parser = Xerces
     [echo] * XSLT processor = Saxon
     [echo] * collator = ICU
     [echo] *****************************************************************
     [echo] #Ant properties
     [echo] #Tue Jul 16 16:14:04 EEST 2024
     [echo] args.outext=htm <--
     [echo] args.grammar.cache=yes
     [echo] args.xml.systemid.set=yes
```

**Reply by JackBaltzer** [reply]
Good catch, but unfortunately I am actually running the correct `-Dargs.outext=htm`  command, I dunno why I wrote the wrong line here. 
If you see the image where i run dita, the correct command is run, and the result is still `args=.outext` not `args.outext=htm`

**Reply by jlacour31** [reply]
I ran the following command using Oxygen Publishing Engine 26.1 (OPE - based on DITA-OT 4.1.2):
```
dita -v -f webhelp-responsive -i path\to\samples\dita\flowers\flowers.ditamap -o path\to\samples\dita\flowers\out\webhelp-responsive -Dargs.outext=htm
```
And I obtained the console from my first comment (with `args.outext=htm`).
Could you indicate which OPE / DITA-OT you are using and which version of Windows?
If you are using DITA-OT indicate which Oxygen WebHelp plugin you have installed.

**Reply by JackBaltzer** [reply]
Thank you for helping with this!


On my server (windows Server 2019 Standard, version 1809) I have tried a fresh reinstall of everything, to be sure I have the correct files.

I downloaded dita-ot-4.1.2.zip from here: [https://github.com/dita-ot/dita-ot/releases/tag/4.1.2](https://github.com/dita-ot/dita-ot/releases/tag/4.1.2)
The contents if the zip is extracted to: C:\Temp\WMSHelp\dita-4.1.2

I downloaded the WebHelp plugin from here [https://www.oxygenxml.com/xml_webhelp/download_oxygenxml_webhelp.html](https://www.oxygenxml.com/xml_webhelp/download_oxygenxml_webhelp.html)
the contents are extracted and copied onto C:\Temp\WMSHelp\dita-4.1.2\plugins

I ran the install command (first I ran the command BEFORE I copied WebHelp, then I ran again after the copy)
![image](https://github.com/user-attachments/assets/9213170b-2fa4-42ba-9e2c-62fc24569e89)


I have the Oxygen WebHelp licensekey.txt in the root of the folder: C:\Temp\WMSHelp\dita-4.1.2
![image](https://github.com/user-attachments/assets/1aef507f-0953-48ab-8ac3-be6d5687400e)



This is the versions of dita and java
![image](https://github.com/user-attachments/assets/6797dc0f-a9e0-4008-9dba-6e6342dc6104)



This is the command I am running, and the initial logs it produces:
![image](https://github.com/user-attachments/assets/e630e195-598a-46fd-b71f-dbf30b0af09e)

**Reply by jlacour31** [reply]
I tried the same steps as yours (except that I don't have Windows Server 2019) and I did get the correct parameter (`args.outext=htm`).
Maybe you could try to use Java 17 instead of 22, this is the version we are using by default on our side.

**Reply by JackBaltzer** [reply]
It is quite strange, I am still getting the `args.=outext` instaed of the expected `args.outext=htm`
![image](https://github.com/user-attachments/assets/c7f56a6f-da98-43d8-875e-59308b1902f9)

**Reply by jlacour31** [reply, marked as answer]
I find out where is the problem, I was using Windows Command while you are using Windows PowerShell.
In order to make it work you need to quote all -D arguments, as explained [here](https://stackoverflow.com/questions/6347985/cannot-run-maven-using-mvn-d-argument-within-microsoft-powershell-but-works).
I tried `'-Dargs.outext=htm'` and the transformation works like a charm!
Sorry that I did not get it earlier, I'm not used to PowerShell.

**Reply by JackBaltzer** [reply]
Ah, I made progress..

I have a smart colleague who told me to try `--args.outext=htm` instaed of `Dargs.outext=htm`  and this seems to work! 
The resulting website does indeed have .htm as extension. 
![image](https://github.com/user-attachments/assets/fe0c9654-6d80-41a1-be64-d5e7068504cf)

**Reply by chrispy-snps** [reply]
@JackBaltzer - even if that form works, I would not use it. The syntax given by @jlacour31 is the correct syntax to use.
